File: src/main/java/kr/co/lunatalk/domain/order/dto/response/OrderItemResponse.java
package kr.co.lunatalk.domain.order.dto.response;

import kr.co.lunatalk.domain.order.domain.OrderItem;

public record OrderItemResponse(
	Long productId,
	String productName,
	Long price,
	Integer quantity,
	Long totalPrice,
	String color,
	String productImageUrl
) {
	public static OrderItemResponse from(OrderItem item) {
		return new OrderItemResponse(
			item.getProductId(),
			item.getProductName(),
			item.getPrice(),
			item.getQuantity(),
			item.getTotalPrice(),
			item.getOptionSnapshot().getColor(),
			null
		);
	}

	public static OrderItemResponse from(OrderItem item, String productImageUrl) {
		return new OrderItemResponse(
			item.getProductId(),
			item.getProductName(),
			item.getPrice(),
			item.getQuantity(),
			item.getTotalPrice(),
			item.getOptionSnapshot().getColor(),
			productImageUrl
		);
	}
}

File: src/main/kotlin/kr/co/lunatalk/domain/order/dto/response/OrderItemResponse.kt
package kr.co.lunatalk.domain.order.dto.response

import kr.co.lunatalk.domain.order.domain.OrderItem

data class OrderItemResponse(
    val productId: Long?,
    val productName: String?,
    val price: Long?,
    val quantity: Int?,
    val totalPrice: Long?,
    val color: String?,
    val productImageUrl: String?
) {
    companion object {
        fun from(item: OrderItem): OrderItemResponse {
            return OrderItemResponse(
                productId = item.productId,
                productName = item.productName,
                price = item.price,
                quantity = item.quantity,
                totalPrice = item.totalPrice,
                color = item.optionSnapshot?.color,
                productImageUrl = null
            )
        }

        fun from(item: OrderItem, productImageUrl: String?): OrderItemResponse {
            return OrderItemResponse(
                productId = item.productId,
                productName = item.productName,
                price = item.price,
                quantity = item.quantity,
                totalPrice = item.totalPrice,
                color = item.optionSnapshot?.color,
                productImageUrl = productImageUrl
            )
        }
    }
}
